File: src/main/java/com/pancisin/bookster/models/Message.java
package com.pancisin.bookster.models;

import java.util.Calendar;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonView;
import com.pancisin.bookster.models.enums.RecipientType;
import com.pancisin.bookster.models.views.Compact;

@Entity
@Table(name = "messages")
public class Message {

	@Id
	@JsonView(Compact.class)
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;
	
	@ManyToOne
	@JsonView(Compact.class)
	private User sender;

	@JsonView(Compact.class)
	@Enumerated(EnumType.STRING)
	private RecipientType recipientType;
	
	@JsonView(Compact.class)
	private Long recipientId;

	@JsonView(Compact.class)
	private String content;

	@JsonView(Compact.class)
	@Column(name = "created", columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP", insertable = false, updatable = false)
	private Calendar created;

	public User getSender() {
		return sender;
	}

	public void setSender(User sender) {
		this.sender = sender;
	}

	public String getContent() {
		return content;
	}

	public void setContent(String content) {
		this.content = content;
	}

	public Calendar getCreated() {
		return created;
	}

	public void setCreated(Calendar created) {
		this.created = created;
	}

	public Long getId() {
		return id;
	}

	public RecipientType getRecipientType() {
		return recipientType;
	}

	public void setRecipientType(RecipientType recipientType) {
		this.recipientType = recipientType;
	}

	public Long getRecipientId() {
		return recipientId;
	}

	public void setRecipientId(Long recipientId) {
		this.recipientId = recipientId;
	}
}

File: src/main/kotlin/com/pancisin/bookster/model/Message.kt
package com.pancisin.bookster.model

import javax.persistence.Entity
import javax.persistence.Table
import javax.persistence.Id
import com.pancisin.bookster.model.enums.RecipientType
import javax.persistence.GeneratedValue
import javax.persistence.ManyToOne
import javax.persistence.Enumerated
import javax.persistence.EnumType
import javax.persistence.Column
import java.util.Calendar
import javax.persistence.GenerationType

@Entity
@Table(name = "messages")
data class Message(

  @Id @GeneratedValue(strategy = GenerationType.AUTO)
  var id: Long? = null,

  @ManyToOne
  var sender: User? = null,

  @Enumerated(EnumType.STRING)
  var recipientType: RecipientType? = null,

  @Column
  var recipientId: Long? = null,

  @Column
  var content: String? = null,

  @Column(name = "created", columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP", insertable = false, updatable = false)
  var created: Calendar = Calendar.getInstance()
)
